ILSpy.ReadyToRun/ReadyToRunLanguage.cs

```diff
@@ -149,7 +152,27 @@ private void Disassemble(ITextOutput output, ReadyToRunReader reader, ReadyToRun
 				for (int i = 0; i < missingBytes; i++)
 					output.Write("  ");
 				output.Write(" ");
-				output.WriteLine(tempOutput.ToStringAndReset());
+				output.Write(tempOutput.ToStringAndReset());
+				int importCellAddress = (int)instr.IPRelativeMemoryAddress;
+				if (instr.IsCallNearIndirect && reader.ImportCellNames.ContainsKey(importCellAddress)) {
+					output.Write(" ;");
+					ReadyToRunSignature signature = reader.ImportSignatures[(int)instr.IPRelativeMemoryAddress];
+					string formattedSignatureString = reader.ImportCellNames[importCellAddress];
+					MethodDefEntrySignature methodDefSignature = signature as MethodDefEntrySignature;
+					MethodRefEntrySignature methodRefSignature = signature as MethodRefEntrySignature;
+					if (methodDefSignature != null) {
+						int methodDefToken = unchecked((int)methodDefSignature.MethodDefToken);
+						output.WriteReference(method.ParentModule.PEFile, MetadataTokens.Handle(methodDefToken), formattedSignatureString);
```

Comment:
The contents of `formattedSignatureString` look a bit strange to me. Would it be possible to use one of the `InstructionOutputExtensions.WriteTo` methods to make the format the same as `ildasm`?

Also it would be nice to display the metadata tokens as comments (just as the IL views does, if the output option is active), linking to the metadata table entries. What do you think?

Reply:
The `InstructionOutputExtensions.WriteTo` worked great :)